qml/CoverPage.qml

```diff
@@ -22,9 +22,24 @@ import Sailfish.Silica 1.0
 CoverBackground {
     id: root
 
-    property bool update: status === Cover.Active || Qt.application.active
+    // This should be deprecated only use visibleChanged.
+    property bool updateCover: status === Cover.Active
+    onUpdateCoverChanged: {
+        if (updateCover) {
+            previewImage.updatePreview()
+        }
+    }
 
-    onUpdateChanged: previewImage.updatePreview()
+    // While peeking both main window and cover window are visible at the
+    // same time. Thus, we can capture from main window when cover
+    // becomes visible.
+    onVisibleChanged: {
+        if (visible) {
+            previewImage.updatePreview()
+        } else if (window.visible) {
+            previewImage.source = ""
```

Comment:
What's the intention for this? Otherwise previewImage content is updated by the updatePreview() function which gets called from multiple places, so this looks a bit inconsistent. Also the fallback icon + file name is set based on image visibility, which is not changed with this.